.github/workflows/test-swaps.yml

```diff
@@ -0,0 +1,140 @@
+name: Test sending swaps transactions
+
+on:
+  workflow_call:
+
+jobs:
+  data-transactions-tests:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 18
+          cache: 'npm'
+          cache-dependency-path: .github/action_scripts/package-lock.json
+
+      - name: Installing node dependencies
+        shell: bash
+        run: |
+          cd .github/action_scripts
+          npm i @stardust-collective/dag4
+          npm i js-sha256
+          npm i axios
+          npm i brotli
+          npm i zod
+
+      - name: Download shared_jars artifacts
+        uses: actions/download-artifact@v4
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          name: shared_jars
+          path: .github/code/shared_jars
+
+      - name: Download Hypergraph Artifact
+        uses: actions/download-artifact@v4
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          name: hypergraph
+          path: .github/code/hypergraph
+
+      - name: Download project template Artifact
+        uses: actions/download-artifact@v4
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          name: project-template-metagraph
+          path: .github/code/metagraphs/project-template-metagraph
+
+      - name: Start Hypergraph
+        uses: "./.github/templates/actions/hypergraph/start"
+        with:
+          INCLUDE_DAG_L1: true
+          DAG_L0_PORT_PREFIX: 90
+          DAG_L1_PORT_PREFIX: 91
+
+      - name: Start Metagraph - Project Template
+        uses: "./.github/templates/actions/metagraph/start"
+        with:
+          INCLUDE_DATA_L1: true
+          METAGRAPH_NAME: project-template-metagraph
+          DL0_PORT_PREFIX: 90
+          ML0_PORT_PREFIX: 80
+          CL1_PORT_PREFIX: 81
+          DL1_PORT_PREFIX: 82
+
+      - name: Check Metagraph cluster
+        run: |
+          cd .github/action_scripts/check_clusters
+          node metagraph-data.js 90 91 80 81 82
+
+      - name: Send transactions
+        run: |
+          cd .github/action_scripts
+          node send_transactions/swaps.js 90 91 80 81 82 ${{ secrets.CI_PRIVATE_KEY }}
```

Comment:
Realizing now we're not parameterizing the workflow template by genesis, which assumes the key pair has not drifted from the template genesis.csv. It might be useful to print the initial balances for debugging any future test failure due to template change.

Reply:
Could you elaborate more on that? Also, it would be great if you will create a separate ticket for that as it's not related to my specific case I'm working on right now.

Reply:
Not required to change here, I think we discussed a couple options in meeting with Marcus -- it should maybe be part of the template changes.

Reply:
Great, will you summarize it somewhere so we can get some context too? Or is it on Marcus?